consensus/state.go

```diff
@@ -1285,6 +1285,14 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 		return
 	}
 
+	// If it is known that this block is valid, prevote it
+	// ProposalBlock already has a nil check above
+	if cs.ValidBlock != nil && bytes.Equal(cs.ValidBlock.Hash(), cs.ProposalBlock.Hash()) {
```

Comment:
May be more idiomatic:
```suggestion
	if cs.ValidBlock != nil && cs.ProposalBlock.HashesTo(cs.ValidBlock.Hash()) {
```